inst/scripts/SEL3D.R

```diff
@@ -0,0 +1,251 @@
+  #will just think it's a new sequence starting from index 1 to its length
+  #so after we blast the subsequence, we need to adjust the qstart and
+  #qend positions by the numbers we stored before the recursion.
+
+  finalOutput$q.start <- finalOutput$q.start + startPos - 1
+  finalOutput$q.end <- finalOutput$q.end + startPos - 1
+
+  #more spooky edge cases
+  minSide <- 0
+  maxSide <- 0
+
+  for(x in 1:length(finalOutputRange)){
+    finalOutputRange[[x]] <- finalOutputRange[[x]] + startPos - 1
+    if(minSide == 0 || min(finalOutputRange[[x]] < minSide)){
+      minSide <- min(finalOutputRange[[x]])
+    }
+
+    if(maxSide == 0 || max(finalOutputRange[[x]] > maxSide)){
+      maxSide <- max(finalOutputRange[[x]])
+    }
+
+  }
+
+
+  #Recursive step: if there are subsequences remaining after our right-most query that
+  #are >50aa in length, blast these as well
+
+  #added this because of random spooky edge cases
+  if(endPos - maxSide > 50){
+    if(!silent) {
+      message(sprintf("running substr from pos %d to %d", maxSide, endPos))
+    }
+    maxFrame <- getPdbIds(substr(AAseq, maxSide - startPos, endPos - startPos),
+                          startPos = maxSide, endPos = endPos, convert = FALSE, silent = silent)
+    finalOutput <- rbind(finalOutput, maxFrame)
+  }
+
+  #Recursive step: if there are subsequences remaining before our leftmost-most query that
+  #are >50aa in length, blast these as well
+  if(minSide - startPos > 50){
+    if(!silent) {
+      message(sprintf("running substr from pos %d to %d", startPos, minSide))
+    }
+    minFrame <- getPdbIds(substr(AAseq, 1, minSide - startPos),
+                          startPos = startPos, endPos = minSide, convert = FALSE, silent = silent)
+    finalOutput <- rbind(finalOutput, minFrame)
+  }
+
+  #We've covered the recursive steps which blast squences book-ending our first and last alignments
+  #now we have to check subsequences between our alignments (if there are multiple alignments) for
+  #the current recursive step
+
+  if(length(finalOutputRange) > 1){ #man when the array is 1 but you need at least 2 but then dont check for it A+ fupan
+    #mysterious stack overflow solution
+    finalOutputRange <- finalOutputRange[order(sapply(finalOutputRange,'[[',1))]
+    #why does this work i dont know i just copied it and bam working code lololol xddd
+    for(y in 1:(length(finalOutputRange) - 1)){
+      leftSide <- max(finalOutputRange[[y]])
+      rightSide <- min(finalOutputRange[[y+1]])
+
+      if(rightSide - leftSide > 50){
+        if(!silent) {
+          message(sprintf("running substr from pos %d to %d", leftSide, rightSide))
+        }
+        interFrame <- getPdbIds(substr(AAseq, leftSide - startPos, rightSide - startPos),
+                                startPos = leftSide, endPos = rightSide, convert = FALSE, silent = silent)
+        finalOutput <- rbind(finalOutput, interFrame)
+      }
+    }
+  }
+
+  return(finalOutput)
```

Comment:
Is it guaranteed that query-sequence coordinates map exactly back on the coding sequence?
Is it guaranteed that this is an ungapped alignment?
Is it guaranteed that alignments don't overlap?